IncludeToolbox/Utils.cs

```diff
@@ -35,23 +35,61 @@ public static string MakeRelative(string absoluteRoot, string absoluteTarget)
             return relativePath;
         }
 
-        public static string GetExactPathName(string pathName)
+        // Shamelessly stolen from https://stackoverflow.com/a/5076517/153079
```

Comment:
This is similar to the previous functionality, but should be more reliable since each part of the path is normalized with GetFileSystemInfos. It's important to get this right, because eventually we rely on string comparisons to determine whether a file is in/under a directory path.

Reply:
Honestly, I didn't hit a problem-case, but I became very concerned about it because I was adding code that uses naked string operations to compare two different paths.

However, there aren't _really_ two functions now, `GetFileSystemCasing` is private, and `GetExactPathName` calls it. I did it that way because the core work is recursive but there's also work you want to do before you start and after you finish, in this case normalizing the separators and adding one to the end of directories respectively.

Reply:
Reading back over the SO comments, I think the old version probably did work. The only thing I'm not 100% sure about is using DI `Name` instead of `FullName`. It probably worked fine, since in both cases the DI is built from `GetFileSystemInfos()`, but I also think the SO people arguing about how to do this did more pedantic/specific testing than you or I.